src/UdApi.ts

```diff
@@ -179,15 +179,19 @@ export default class Udapi extends NamingService {
   }
 
   async registryAddress(domain: string): Promise<string> {
-    const serviceName = findNamingServiceName(domain);
+    const tld = domain.split('.').pop();
+    if (tld === undefined) {
+      throw new ResolutionError(ResolutionErrorCode.UnsupportedService, {
+        namingService: tld,
+      });
+    }
 
-    if (serviceName === 'CNS') {
-      return NetworkConfig.networks[1].contracts.Registry.address;
-    } else if (serviceName === 'ZNS') {
+    if (tld === 'crypto') {
+      return UnsConfig.networks[1].contracts.CNSRegistry.address;
```

Comment:
We can't know for sure what network is mentioned when the user use Resolution API. It worth adding znsNetwork and cns(uns)Network properties in this class and set them explicitly